sympy/combinatorics/rewritingsystem_fsm.py

```diff
@@ -0,0 +1,62 @@
+class State(object):
+    """A representation of a state managed by a ``StateMachine``.
+
+    Attributes:
+        name (str): State name which is also assigned to the Machine.
+        transisitons (OrderedDict): Represents all the transitions of the state object.
+        is_start (boolean): To mark if the state is a start state.
+        is_dead (boolean): To mark if the state is a dead state.
+        is_accept (boolean): To mark if the state is an accept state.
+    """
+
+
+    def __init__(self, name, is_start=False, is_dead=False, is_accept=False):
+        self.name = name
+        self.is_start = is_start
+        self.is_dead = is_dead
+        self.is_accept = is_accept
+        self.transitions = {}
+        self.state_type = None
+        if is_start:
+            self.state_type = "start"
+        elif is_accept:
+            self.state_type = "accept"
+        elif is_dead:
+            self.state_type = "dead"
+
+    def add_transition(self, letter, state):
+        """
+        This method is to add a transition from the current state to a new state.
```

Comment:
A note on docstrings. Have a look at [this wiki page](https://github.com/sympy/sympy/wiki/Writing-documentation). In particular, it suggests that [these conventions](https://www.python.org/dev/peps/pep-0257/) should be followed for docstring writing. Here's a quote from there:

> The docstring is a phrase ending in a period. It prescribes the function or method's effect as a command ("Do this", "Return that"), not as a description; e.g. don't write "Returns the pathname ...".

So this docstring should be something like `Add a transition from the current state to a new state`. And the same applies to all other docstrings here (except classes I suppose; class docstring tend to be nouns with descriptions).

Reply:
You still need to change the docstrings to the right format

Reply:
Yes, I will push it along with these changes